src/java/org/apache/cassandra/service/consensus/TransactionalMode.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.cassandra.service.accord.IAccordService;
+
+/*
+ * Configure the transactional behavior of a table. Enables accord on a table and defines how it mixes with non-serial writes
+ *
+ * For Accord transactions to function correctly when mixed with non-SERIAL writes it's necessary for the writes to occur through Accord.
+ *
+ * Accord will also use this configuration to determine what consistency level to perform its reads
+ * at since it will need to be able to read data written at non-SERIAL consistency levels.
+ *
+ * BlockingReadRepair will also use this configuration to determine how BRR mutations are applied. For migration
+ * and accord the BRR mutations will be applied as Accord transactions so that BRR doesn't expose Accord to
+ * uncommitted Accord data that is being RRed. This can occur when Accord has applied a transaction at some, but not
+ * all replica since Accord defaults to asynchronous commit.
+ *
+ * By routing repairs through Accord it is guaranteed that the Accord derived contents of the repair have already been applied at any
+ * replica where Accord applies the transaction. This also prevents BRR from breaking atomicity of Accord writes.
+ *
+ * If they are not written through Accord then reads through Accord will be required to occur at
+ * consistency level compatible with the non-serial writes preventing single replica reads from being performed
+ * by Accord. It will also require Accord to perform read repair of non-serial writes.
+ *
+ * Even then there is the potential for Accord to inconsistently execute transactions at different replicas
+ * because different coordinators for an Accord transaction may encounter different non-SERIAL write state and
+ * race to commit different outcomes for the transaction.
+ *
+ * This is different from Paxos because Paxos performs consensus on the actual values to be applied so recovery
+ * coordinators will always produce a consistent state when applying a transaction. Accord performs consensus on
+ * the execution order of transaction and different coordinators witnessing different states not managed by Accord
+ * can produce multiple outcomes for a transaction.
+ *
+ * // TODO to safely migrate you would have to route all writes through Accord with the current implementation
+ * // We could do it by range instead in the migration version, but then we need to know when all in flight writes
+ * // are done before marking a range as migrated. Would waiting out the timeout be enough (timeout bugs!)?
+ */
+public enum TransactionalMode
+{
+    off(false, false, false, false),
+
+    /*
+     * Execute writes through Cassandra via StorageProxy's normal write path. This can lead Accord to compute
+     * multiple outcomes for a transaction that depends on data written by non-SERIAL writes.
+     */
+    unsafe(true, false, false, false),
+
+    /*
+     * Allow mixing of non-SERIAL writes and Accord, but still force BRR through Accord
```

Comment:
```suggestion
     * Allow mixing of non-SERIAL writes and Accord, but still force BRR through Accord.
     * This mode makes it safe to perform non-SERIAL or SERIAL reads of Accord data, but unsafe
     * to write data that Accord may attempt to read.
```

Reply:
I got this slightly wrong. It makes it safe to perform SERIAL writes/reads and non-SERIAL reads. The same is true of the "unsafe" mode.